code/game/objects/items/melee/baton.dm

```diff
@@ -660,17 +660,27 @@
 	sparkler = new (src)
 
 /obj/item/melee/baton/security/cattleprod/attackby(obj/item/item, mob/user, params)//handles sticking a crystal onto a stunprod to make a teleprod
-	if(!istype(item, /obj/item/stack/ore/bluespace_crystal))
-		return ..()
+	. = ..()
 	if(!cell)
-		var/obj/item/stack/ore/bluespace_crystal/crystal = item
-		var/obj/item/melee/baton/security/cattleprod/teleprod/prod = new
-		remove_item_from_storage(user)
-		qdel(src)
-		crystal.use(1)
-		user.put_in_hands(prod)
-		to_chat(user, span_notice("You place the bluespace crystal firmly into the igniter."))
-	else
+		if(istype(item, /obj/item/stack/ore/bluespace_crystal))
+			var/obj/item/stack/ore/bluespace_crystal/crystal = item
+			var/obj/item/melee/baton/security/cattleprod/teleprod/prod = new
+			remove_item_from_storage(user)
+			qdel(src)
+			crystal.use(1)
+			user.put_in_hands(prod)
+			to_chat(user, span_notice("You place the bluespace crystal firmly into the igniter."))
+
+		else if(istype(item, /obj/item/stack/telecrystal))
+			var/obj/item/stack/telecrystal/crystal = item
+			var/obj/item/melee/baton/security/cattleprod/telecrystalprod/prod = new
+			remove_item_from_storage(user)
+			qdel(src)
+			crystal.use(1)
+			user.put_in_hands(prod)
+			to_chat(user, span_notice("You place the telecrystal firmly into the igniter."))
```

Comment:
This looks like a lot (if not all) can be generalized, instead of doing the same thing twice.

Reply:
consequence of sleep deprivation, I'll tinker with it.

Reply:
I've had an honest attempt at generalizing it, but the stack stuff was giving me a headache and I might need a bit of advice to improve that, if at all possible. Have a look and see what you think, also at the stuff I was trying in discord.